### PR TITLE
[react-native] ran dprint fmt

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
👋 Hi! This PR is meant to let you know that we're going to be applying formatting automatically to all DefinitelyTyped packages[^1]. This PR is split out of:

* https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/65993: the discussion that includes backing context and our planned approach
* https://github.com/DefinitelyTyped/DefinitelyTyped/pull/66235: onboards the repository to [dprint](https://dprint.dev)

This package is particularly huge (>1MB), so we're tagging you in an individual PR as a heads up. Please post here to let us know if you have any workflow needs that auto-formatting might disrupt[^2].

Cheers! 💖 

```shell
time npx dprint fmt types/react-native/**/*.t*
Formatted 282 files.
npx dprint fmt types/react-native/**/*.t*  2.16s user 0.38s system 282% cpu 0.898 total
```

[^1]: Note that formatting will be applied as a commit to `master` after each PR. PRs won't be blocked on any CI check for proper formatting. See #65993 for more information.
[^2]: For example, if this package is fully auto-generated (rather than manually edited), subsequent PRs might have a lot of noise from formatting differences. We can exclude it from formatting in that case.

### CODE REVIEW

File: types/react-native/test/globals.tsx [hunk 1]
```diff
@@ -1,108 +1,150 @@
...
-    }, 0, 100, 'missing-arg');
+    handle = setInterval(
+        (arg1: number) => {
+            console.log("arg1", arg1);
+            // @ts-expect-error
+        },
+        0,
+        100,
+        "missing-arg",
+    );
```
Comment (JoshuaKGoldberg) [inline types/react-native/test/globals.tsx:44]: At first glance this looked like a bug in dprint... but I don't see a better way than how dprint is going. Will touch up manually.

**Review by lunaleaps — APPROVED** [review]
I think this is fine. Note these types are copied from https://github.com/facebook/react-native/tree/main/packages/react-native/types but as long as we can re-format here that should be fine.

**Comment by JoshuaKGoldberg** [comment]
Aha! That is exactly what we're looking for with "auto-generated" packages (copying from an external source counts in this context). I'll close this PR and instead add `"./types/react-native"` to the list of excluded packages in #66235. Thanks @lunaleaps!

I also spy a `tslint.json` in that directory - just a heads up that we'll soon move to only having ESLint files. I'll set a reminder for myself to check in on it in a month & offer help if it'd be useful. Cheers!

**Comment by JoshuaKGoldberg** [comment]
Just following up: https://github.com/react-native-community/discussions-and-proposals/issues/719

**Pull request closed without merging**